examples/inshop/convert_inshop.py

```diff
@@ -131,12 +117,27 @@ def build_inshop_df(
         df=df, ratio_th=bboxes_aspect_ratio_to_fix, fix_train=fix_train_bboxes, fix_val=fix_val_bboxes
     )
 
-    le = preprocessing.LabelEncoder()
-    df["category"] = le.fit_transform(df["category_name"])
+    df["category"] = df_part["path"].apply(lambda x: f"{x.parent.parent.parent.name}/{x.parent.parent.name}")
+
+    df = df[["label", "path", "split", "is_query", "is_gallery", "x_1", "x_2", "y_1", "y_2", "category"]]
 
-    df = df[
-        ["label", "path", "split", "is_query", "is_gallery", "x_1", "x_2", "y_1", "y_2", "category", "category_name"]
-    ]
+    # check stat
+    assert df["path"].nunique() == len(df) == 52712
+    assert df["label"].nunique() == 7982
+    assert set(df["label"].astype(int).tolist()) == set(list(range(1, 7982 + 1)))
+
+    mask_non_single_images = df.groupby("label").label.transform("count") > 1
```

Comment:
this step have to launch as last step

Dropping bad boxes may lead to label with single instance

Reply:
done